test/test_helper.rb

```diff
@@ -110,6 +113,58 @@ def fake_machine(options = {})
   machine
 end
 
+class FakeHTTPProxyServer < WEBrick::HTTPProxyServer
+  def initialize(url, config)
+    @url = url
+    super(config)
+  end
+
+  def service(req, res)
+    req.attributes['url'] = @url
+    super(req, res)
+  end
+end
+
+module WEBrick
```

Comment:
> Tried that solution but it seems to be not working. 

Have you followed the call chain. If the IO instance makes it into the WebRick class, everything looks as it should be. The question is why it does not end up there? If you attach the debugger, you should get the answer pretty much immediately.